internal/telemetry/segment.go

```diff
@@ -117,21 +118,34 @@ func LogShellDurationEvent(eventName string, startTime string) error {
 	return nil
 }
 
-// UserIDFromGithubUsername hashes the github username and produces a 64-char string as userID.
-// Returns an empty string if no github username is found.
+// UserIDFromGithubUsername returns a uuid string if the user has authenticated with github.
+// If not authenticated, or there's an error, then an empty string is returned, which segment
+// would treat as logged-out or anonymous user.
 func UserIDFromGithubUsername() string {
 	username, err := openssh.GithubUsernameFromLocalFile()
 	if err != nil || username == "" {
 		return ""
 	}
 
-	const salt = "d6134cd5-347d-4b7c-a2d0-295c0f677948"
-	mac := hmac.New(sha256.New, []byte(salt))
+	// hashFromUsername hashes the github username and produces a 64-char string as userID.
+	// Returns an empty string if no github username is found.
+	hashFromUsername := func(username string) string {
 
-	const githubPrefix = "github:"
-	mac.Write([]byte(githubPrefix + username))
+		const salt = "d6134cd5-347d-4b7c-a2d0-295c0f677948"
+		mac := hmac.New(sha256.New, []byte(salt))
 
-	return hex.EncodeToString(mac.Sum(nil))
+		const githubPrefix = "github:"
+		mac.Write([]byte(githubPrefix + username))
+
+		return hex.EncodeToString(mac.Sum(nil))
+	}
+
+	hash := hashFromUsername(username)
+	uid, err := uuid.FromBytes([]byte(hash)[:16])
```

Comment:
I don't think this will create a UUID with a valid version/variant since all of the bytes are random. This might not matter for Sentry though (I'm not sure why they want a UUID specifically). `uuid.NewHash` might be a better way of hashing the username into a UUID.

Reply:
Segment (not sentry) doesn't ask for a UUID. They do recommend choosing a `UserID` that will be stable for the same user, so we can track the same user over time.

Prior to this PR, I was using just the raw 64-char hash. But I felt that would be too large, and unstructured in case we choose to use that value later on in some internal database (hypothetically). So, I am exploring using a uuid type instead.

Thanks for pointing out the problem about the invalid version/variant. Will look into that.